internal/provider/provider.go

```diff
@@ -109,59 +71,101 @@ func (p *provider) login(ctx context.Context, client *api.Client, vaultKubernete
 	return secret.Auth.ClientToken, nil
 }
 
-func (p *provider) getSecret(ctx context.Context, client *api.Client, secret config.Secret) (content string, err error) {
-	p.logger.Debug("vault: getting secrets from vault...")
-
-	s := regexp.MustCompile("/+").Split(secret.ObjectPath, 3)
-	if len(s) < 3 {
-		return "", fmt.Errorf("unable to parse secret path %q", secret.ObjectPath)
+func ensureV1Prefix(s string) string {
+	switch {
+	case strings.HasPrefix(s, "/v1/"):
+		return s
+	case strings.HasPrefix(s, "v1/"):
+		return "/" + s
+	case strings.HasPrefix(s, "/"):
+		return "/v1" + s
+	default:
+		return "/v1/" + s
 	}
-	secretPrefix := s[1]
-	secretSuffix := s[2]
+}
 
-	secretMountType, secretMountVersion, err := p.getMountInfo(ctx, client, secretPrefix)
-	if err != nil {
-		return "", err
+func generateRequest(client *api.Client, secret config.Secret) (*api.Request, error) {
+	secretPath := ensureV1Prefix(secret.SecretPath)
```

Comment:
I wonder if splitting on `/` would be better and checking what the first value is. Then we can prepend the API version cleanly.

Reply:
I had a little play with this idea but personally found the logic both harder to follow and make robust, but perhaps I don't fully understand your idea. If you think this could simplify it, do you have a little code sample?